Riot/Modules/Room/RoomViewController.m

```diff
@@ -3195,18 +3196,19 @@ - (IBAction)onButtonPressed:(id)sender
         [self showExpandedHeader:NO];
         // Dismiss potential keyboard.
         [self dismissKeyboard];
-        
-        MXKRoomDataSource *roomDataSource;
+
         // Jump to the last unread event by using a temporary room data source initialized with the last unread event id.
-        roomDataSource = [[RoomDataSource alloc] initWithRoomId:self.roomDataSource.roomId initialEventId:self.roomDataSource.room.accountData.readMarkerEventId andMatrixSession:self.mainSession];
-        [roomDataSource finalizeInitialization];
-        
-        // Center the bubbles table content on the bottom of the read marker event in order to display correctly the read marker view.
-        self.centerBubblesTableViewContentOnTheInitialEventBottom = YES;
-        [self displayRoom:roomDataSource];
-        
-        // Give the data source ownership to the room view controller.
-        self.hasRoomDataSourceOwnership = YES;
+        [RoomDataSource loadRoomDataSourceWithRoomId:self.roomDataSource.roomId initialEventId:self.roomDataSource.room.accountData.readMarkerEventId andMatrixSession:self.mainSession onComplete:^(id roomDataSource) {
```

Comment:
We can add weak self reference in block to not retain self if process take a lot of time